### PR TITLE
QPPCT-566: API acceptance tests

### PR DESCRIPTION
**Pull request by halprin**
### Information
- JIRA story QPPCT-566.

### Changes proposed in this PR:
- Added acceptance tests for the CPC+ APIs.
- Added acceptance tests for the QRDA3 conversion API.
- Deleted the health acceptance test because the existence of the health integration test.

### Checklist 
- [x] All JUnit tests pass (`mvn clean verify`).
- [x] New unit tests written to cover new functionality.
- [x] Added and updated JavaDocs for non-test classes and methods.
- [x] No local design debt. Do you feel that something is "ugly" after your changes?
- [x] Updated documentation (`README.md`, etc.) depending if the changes require it.

### CODE REVIEW

File: rest-api/src/test/java/gov/cms/qpp/conversion/api/acceptance/CpcApiAcceptance.java [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+			.auth().oauth2(createCpcJwtToken())
+			.get(CPC_FILE_API_PATH + firstFileId)
+			.then()
+			.statusCode(200)
+			.contentType("application/xml");
+	}
+
+	@Test
+	@Tag("acceptance")
+	void testNoSecurityMarkFileUnprocessed() {
```
Comment (sfradkin) [inline rest-api/src/test/java/gov/cms/qpp/conversion/api/acceptance/CpcApiAcceptance.java]: Can we get a quick rename here?  You're marking the file as processed, not unprocessed.

File: rest-api/src/test/java/gov/cms/qpp/conversion/api/acceptance/CpcApiAcceptance.java [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+	void testGetFile() {
+
+		String firstFileId = getFirstUnprocessedCpcFileId();
+
+		given()
+			.auth().oauth2(createCpcJwtToken())
+			.get(CPC_FILE_API_PATH + firstFileId)
+			.then()
+			.statusCode(200)
+			.contentType("application/xml");
```
Comment (sfradkin) [inline rest-api/src/test/java/gov/cms/qpp/conversion/api/acceptance/CpcApiAcceptance.java]: Do we want to go as far as verifying any part of the actual content returned?

File: rest-api/src/test/java/gov/cms/qpp/conversion/api/acceptance/QrdaApiAcceptance.java [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+		long afterObjectCount = getS3ObjectCount();
+		long afterDynamoCount = getDynamoItemCount();
+
+		assertThat(afterObjectCount).isEqualTo(beforeObjectCount + 4);
+		assertThat(afterDynamoCount).isEqualTo(beforeDynamoCount + 1);
+	}
+
+	private long getS3ObjectCount() {
+
+		ObjectListing objectListing = s3Client.listObjects("flexion-qpp-conversion-tool-pii-convrtr-audt-test-us-east-1");
```
Comment (sfradkin) [inline rest-api/src/test/java/gov/cms/qpp/conversion/api/acceptance/QrdaApiAcceptance.java]: Please create a constant for the bucket name.

File: rest-api/src/test/java/gov/cms/qpp/conversion/api/acceptance/QrdaApiAcceptance.java [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+		while(objectListing.isTruncated()) {
+			objectListing = s3Client.listNextBatchOfObjects(objectListing);
+			objectCount += objectListing.getObjectSummaries().size();
+		}
+
+		return objectCount;
+	}
+
+	private long getDynamoItemCount() {
+		return dynamoClient.scan("qpp-qrda3converter-test-metadata", Lists.newArrayList("Uuid")).getCount();
```
Comment (sfradkin) [inline rest-api/src/test/java/gov/cms/qpp/conversion/api/acceptance/QrdaApiAcceptance.java]: Please create a constant for the dynamodb table name.

File: rest-api/src/test/java/gov/cms/qpp/conversion/api/acceptance/CpcApiAcceptance.java (hunk 1 above)
Comment (halprin) [reply rest-api/src/test/java/gov/cms/qpp/conversion/api/acceptance/CpcApiAcceptance.java]: 🤦‍♂️ Oops.

File: rest-api/src/test/java/gov/cms/qpp/conversion/api/acceptance/CpcApiAcceptance.java (hunk 2 above)
Comment (halprin) [reply rest-api/src/test/java/gov/cms/qpp/conversion/api/acceptance/CpcApiAcceptance.java]: We could possibly verify that the file has the part that identifies it as a CPC+.

File: rest-api/src/test/java/gov/cms/qpp/conversion/api/acceptance/CpcApiAcceptance.java [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+	@Test
+	@Tag("acceptance")
+	void testNoSecurityForUnprocessedFiles() {
+		get(CPC_UNPROCESSED_FILES_API_PATH)
+			.then()
+			.statusCode(403);
+	}
+
+	@Test
+	@Tag("acceptance")
```
Comment (omnivion) [inline rest-api/src/test/java/gov/cms/qpp/conversion/api/acceptance/CpcApiAcceptance.java]: Let's add a meta annotation @AcceptanceTest which has @Test and @Tag("acceptance")

**Pull request merged**